src/enterPortalSubtask.c

```diff
@@ -66,3 +84,120 @@ void sub_0804AB24(void) {
 void Subtask_PortalCutscene(void) {
     gUnk_080D412C[gArea.filler[8]]();
 }
+void sub_0804AB70(void) {
+    const u8* ptr;
+    RoomControls* controls;
+    u32 portalId;
+
+    DispReset(0);
+    gScreen.lcd.displayControl = 0x1240;
+    gScreen.bg1.control = 0x9e83;
+    controls = &gRoomControls;
+    gScreen.bg1.xOffset = 0;
+    controls->scroll_x = 0;
+    gScreen.bg1.yOffset = 0;
+    controls->scroll_y = 0;
+    portalId = gArea.curPortalType;
+    if ((portalId == 2) && (gUnk_02032EC0.unk_1c.area != 2)) {
+        portalId = 3;
+    }
+    ptr = &gUnk_080D4138[portalId * 2];
+    LoadPaletteGroup(ptr[0]);
+    LoadGfxGroup(ptr[1]);
+    MemCopy(&gUnk_02017700, gUnk_02017700 + 0x240, 0x20);
+    gUsedPalettes |= 0x200000;
+    EraseAllEntities();
+    LoadRoomEntityList((EntityData*)gUnk_080D4110[portalId]);
+    ResetSystemPriority();
+    gArea.filler3[0]++;
+    SetFade(4, 8);
+}
+
+void sub_0804AC1C(void) {
+    s32 tmp;
+
+    UpdateEntities();
+    if (gUnk_02018EB0.unk_14 != NULL) {
+        tmp = gRoomControls.scroll_y + 0x50 - gUnk_02018EB0.unk_14->y.HALF.HI - gUnk_02018EB0.unk_14->z.HALF.HI;
+        if (tmp != 0) {
+            tmp = gRoomControls.scroll_y = gRoomControls.scroll_y - tmp;
+            if (tmp * 0x10000 < 0) {
+                gRoomControls.scroll_y = 0;
+            }
+            if (0x9f < gRoomControls.scroll_y) {
+                gRoomControls.scroll_y = 0x9f;
+            }
+        }
+    }
+    gScreen.bg1.xOffset = gRoomControls.scroll_x;
+    gScreen.bg1.yOffset = gRoomControls.scroll_y;
+    FlushSprites();
+    DrawEntities();
+    CopyOAM();
+    if ((gUnk_02018EB0.unk_1b != 0) || sub_0804ACA8()) {
+        gArea.filler3[0]++;
+        SetFade(7, 0x10);
+    }
+}
+
+bool32 sub_0804ACA8(void) {
+    if ((gInput.newKeys & (R_BUTTON | B_BUTTON)) == 0) {
+        return 0;
+    } else {
+        return sub_0804AD18();
+    }
+}
+
+void sub_0804ACC8(void) {
+    if (gFadeControl.active == 0) {
+        SetGlobalFlag(gArea.curPortalType + ENTRANCE_0);
+        sub_0804AD6C((RoomControls*)&gUnk_02032EC0.unk_1c);
+        sub_080A71F4(0);
+    }
+}
+
+void RespawnAsMinish(void) {
+    if (sub_0804AD18()) {
+        sub_0804AD6C(&gRoomControls);
+        SetInitializationPriority();
+        SoundReq(SFX_F8);
+    }
+}
+
+bool32 sub_0804AD18(void) {
+    switch (gArea.curPortalType) {
+        case 0:
+        case 1:
+        case 2:
+            if (!CheckGlobalFlag(ENTRANCE_0 + gArea.curPortalType)) {
+                return FALSE;
+            }
+            // else: return TRUE implicitely, because it's stored in r0. But does not match if returning explicitely.
```

Comment:
mark this as //!@bug?

Reply:
I don't think it's a bug, though. There probably is a way to match it with an additional `return TRUE` there or somehow returning the result of the function directly, I've just not found it yet. Maybe a todo instead?